File: backend/app/core/roles.py
"""Shared role groupings.

Several routers each keep their own copy of these tuples (analytics, companies,
communications, daily_updates, officers). New code should import from here; the
existing copies are left alone deliberately, so that consolidating them is its
own small change rather than noise inside a feature.
"""

from app.models.user import UserRole

# The heads: everyone who sees the whole college rather than their own slice.
LEADERSHIP_ROLES = (
    UserRole.SUPER_ADMIN,
    UserRole.PRINCIPAL,
    UserRole.PRO_CHANCELLOR,
    UserRole.DEPUTY_PRO_CHANCELLOR,
)

# The people on the ground, who file daily updates and own company allocations.
FILER_ROLES = (
    UserRole.PLACEMENT_OFFICER,
    UserRole.DEPARTMENT_COORDINATOR,
)

# Everyone who works in the staff app. Students are deliberately not here.
STAFF_ROLES = LEADERSHIP_ROLES + FILER_ROLES

File: frontend/src/lib/roles.ts
import type { UserRole } from '@/types'

/**
 * The heads: everyone who sees the whole college rather than their own slice.
 *
 * Mirrors `LEADERSHIP_ROLES` in `backend/app/core/roles.py`, which is what the
 * leadership-only endpoints actually enforce. This copy exists so the UI can
 * avoid *offering* something the API would refuse — it is not the check. The
 * server's is.
 */
export const LEADERSHIP_ROLES: UserRole[] = [
  'super_admin',
  'principal',
  'pro_chancellor',
  'deputy_pro_chancellor',
]

export function isLeadership(role?: UserRole | null): boolean {
  return !!role && LEADERSHIP_ROLES.includes(role)
}
